src/client.rs

```diff
@@ -1280,6 +1280,48 @@ impl<T: Read + Write> Session<T> {
         .and_then(|lines| parse_id_seq(&lines, &mut self.unsolicited_responses_tx))
     }
 
+    /// The [`SETQUOTA` command](https://datatracker.ietf.org/doc/html/rfc2087#section-4.1)
+    ///
+    /// Updates the resource limits for a mailbox. Any previous limits for the named quota
+    /// are discarded.
+    /// Returns the updated quota
+    pub fn set_quota<S: AsRef<str>>(
+        &mut self,
+        quota_root: S,
```

Comment:
Let's use `impl AsRef<str>` here too.

Reply:
What is the difference between `<S: AsRef<str>>` and `impl AsRef<str>`? Is there some documentation on when to use which variant? (I don't know what to web-search for, I'm watching some repos to learn best practices and I haven't seen `impl AsRef` before, only the other variant.)

Reply:
`<>` generics are specifiable using the turbofish (e.g., `client.set_quota::<String>(...)`), whereas `impl Trait` is not. That may seem like a restriction, but in practice it allows you to ensure that "important" generics are made more explicit, whereas "simple" generics (like `impl AsRef<str>`) sort of fade into the background.

Reply:
Thanks for the explanation @jonhoo. Will see to integrate this in https://rust-unofficial.github.io/patterns/